src/main/common/vector.c

```diff
@@ -96,6 +96,17 @@ vector2_t *vector2Normalize(vector2_t *result, const vector2_t *v)
     }
 }
 
+// rotate 2d vector by angle
+// angle is in radians and counterclockwise
+vector2_t* vector2Rotate(vector2_t *result, const vector2_t *v, const float angle)
```

Comment:
Should the comment indicate the sign?
ie is this correct?
`// angle is in radians and positive values rotate counterclockwise
`

Reply:
This is simple vector rotation, as used in math. https://en.wikipedia.org/wiki/Rotations_and_reflections_in_two_dimensions

In right-handed system, angle is counterclockwise (it is always from X to Y)

> For pilots, positive yaw stick results in right yaw. But I notice in Betaflight that positive yaw gyro values mean left turn, but positive attitude.values.yaw means a rightwards turn. What we should do here needs to be resolved.

Well, this is heritage from multiwii or even more ancient times. You want right stick movement to cause right yaw. And RC transmitter convention is that value increases to the right. So someone added `* -1` to make it work this way.

Gyro uses normal right-handed coordinate system, Z axis is up and positive rotation is counterclockwise.

I don't think that we can resolve this one. It is here for too long, causing headaches for over a decade. For nowm, we can only document it, put warning in relevant places and live with it.

(one aspect of left-handed NEU system used may be that this isn't too visible in IMU. It may be bigger problem with ENU).